var/spack/repos/builtin/packages/connectome-tools/package.py

```diff
@@ -13,21 +13,21 @@ class ConnectomeTools(PythonPackage):
     git = "git@bbpgitlab.epfl.ch:nse/connectome-tools.git"
 
     version('develop', branch='main')
-    version('0.6.1', tag='connectome-tools-v0.6.1')
+    version('0.6.2', tag='connectome-tools-v0.6.2')
 
     depends_on('python@3.8:', type=('build', 'run'))
     depends_on('py-setuptools', type=('build', 'run'))
 
     depends_on('py-click@7.0:8.999', type=('build', 'run'))
     depends_on('py-equation@1.2:', type=('build', 'run'))
     depends_on('py-joblib@1.0.1:', type=('build', 'run'))
-    depends_on('py-jsonschema@3.2.0:3.999', type=('build', 'run'))
+    depends_on('py-jsonschema@3.2.0:4.999', type=('build', 'run'))
```

Comment:
Can we just add these limits when there is a *known* inconsistency?

I'd like to add that limiting versions this way also makes it hard to substitute in a `@develop` build for a dependency.

Reply:
If we execute periodically the automated tests it should be fine to relax the upper dependency.
However, `setup.py` and `package.py` should stay in sync, so in this case it cannot be done until the next release.
In general, I think that the upper limit is usually added for dependencies that are known to be problematic when a major and incompatible version is released (bluepy for example, but also some versions of dask).

Reply:
`dask` switched to some weird year.month release rhythm. I'm not sure how well that leads to any kind of behavior inference.

I'm fine with a few of these, but for example here this seems a bit overkill.

Reply:
Adding more info about dask, I've read this statement from the developers, before switching to CalVer:
> Dask doesn't use semantic versioning. We use a rolling release cycle similar to numpy, pandas, scikit-learn, and other projects in this space.
>
> Due to the large API surface area, pretty much every release would technically bump the major version.

For this reason I think there is no guarantee of backward compatibility even with the current versioning.

Reply:
If there's no hard failure mode for these inconsistencies, I'd also say that maybe we should not codify this in the dependencies, but state a global project wide preference in `deploy/configs/packages.yaml`. I will be doing this for the new deployment [like so](https://github.com/BlueBrain/spack/blob/merge-upstream/deploy/configs/packages.yaml#L81-L86):
```
  py-numpy:
    version: [1.19.5]  # thanks, tensorflow!
  py-tables:
    variants: +lzo  # leads to compile errors otherwise
  py-setuptools:
    version: [57.4.0]  # last one to provide use_2to3, required by some software
```

By pinning the versions globally / stating preferences this way, we will also avoid a bunch of unnecessary rebuilds with slightly different versions (see `setuptools`… :( )